### DISCUSSION TITLE
Check When Progressive Load is Finished

### DISCUSSION THREAD

**Discussion by lewysigns** [anvilistas/tabulator#97] [Q&A] [answered]
if you're doing a progressive load , options as follows:

```python
    self.tabulator.options = {
        "app_table": app_tables.my_table,
        "height" 400,
        "pagination": False,
        "progressive_load": "load", # enable progressive loading
        "progressive_load_delay": 200, # wait 200 milliseconds between each request
```

Is there a way to tell when that progressive load is complete?

**Comment by s-cork** [comment, marked as answer]
I don't know if you can - there doesn't seem to be an event triggered by tabulator to signal this

what's the use case?

**Reply by lewysigns** [reply]
I want to load a large table and then make a "download" button appear once the table is fully loaded. More of a user management strategy.

I told my user to "wait until it stops spinning", but I'd like to set it up where I don't have to explain to the user.

**Reply by lewysigns** [reply]
Just thought, I could have a timer set up that checks the length of the tabulator data and stops when the length repeats.

**Reply by s-cork** [reply]
I can create a custom event when the data is finished loading
just need a sensible name for the event

then you can do
`self.tabulator_1.on(<event-name>, self....)`

pick a name

**Reply by s-cork** [reply]
on dev you should be able to do 

```python

self.tabulator_1.on("app_table_data_loaded", self....)

```

**Reply by lewysigns** [reply]
I could take a wack at it if you point me in the right direction.

How about "progressive_load_complete" or "load_complete" if thats too long?
